byudml/imputer/random_sampling_imputer.py

```diff
@@ -1,29 +1,44 @@
 import numpy as np
+import typing
 
-from d3m import container
+from d3m import container, exceptions as d3m_exceptions
 from d3m.metadata import base as metadata_base, hyperparams, params
 from d3m.primitive_interfaces.base import CallResult
 from d3m.primitive_interfaces.unsupervised_learning import UnsupervisedLearnerPrimitiveBase
 
-from byudml import __version__ as __package_version__
 from byudml import __imputer_path__, __imputer_version__
+from byudml import __version__ as __package_version__
 
 
 Inputs = container.pandas.DataFrame
 Outputs = container.pandas.DataFrame
 
 
-class Params(params.Params):
+class Hyperparams(hyperparams.Hyperparams):
+    drop_missing_values = hyperparams.UniformBool(
+        default=True,
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description='Determines whether to drop columns containing missing values.'
+    )
+    how = hyperparams.Enumeration[str](
+        values=['all', 'any'],
+        default='all',
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description='Determines how to drop missing values. If "all", drops columns where all values are missing. If "any", drops columns where any values are missing (note no imputation is performed).'
+    )
```

Comment:
I like the semantics of these hyperparameters. Also, it's nice that the `how` param let's this primitive double as a drop NaNs primitive.